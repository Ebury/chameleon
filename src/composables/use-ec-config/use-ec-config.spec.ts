import { mount } from '@vue/test-utils';

import useEcConfig, { CHAMELEON_CONFIG_KEY } from './use-ec-config';

describe('useEcConfig', () => {
  function mountUseEcConfig(mountOpts = {}) {
    const Component = {
      setup() {
        const config = useEcConfig();
        return { config };
      },
      template: '<div>{{ config }}</div>',
    };

    return mount(Component, {
      global: {
        provide: {
          [CHAMELEON_CONFIG_KEY]: 'mockedValue',
        },
      },
      ...mountOpts,
    });
  }

  it('should retrieve the config', () => {
    const wrapper = mountUseEcConfig();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should retrieve default config if not provided', () => {
    const wrapper = mountUseEcConfig({
      global: {},
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
