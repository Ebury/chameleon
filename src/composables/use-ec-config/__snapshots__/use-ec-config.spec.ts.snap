// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useEcConfig > should retrieve default config if not provided 1`] = `
<div>
  {
  "sensitiveClass": "my-sensitive-content-test-class",
  "iconsStaticPrefix": ""
}
</div>
`;

exports[`useEcConfig > should retrieve the config 1`] = `
<div>
  mockedValue
</div>
`;
